source/fundamentals/monitoring.txt

```diff
@@ -0,0 +1,406 @@
+   commands. For a full list of commands the driver can't monitor,
+   see the 
+   :spec:`MongoDB command monitoring specification </command-monitoring/command-monitoring.rst#security>`.
+
+Example
+^^^^^^^
+
+This example shows how to make a timer for database commands. The timer makes it so
+that every time the driver executes a command on your MongoDB instance
+you see a message letting you know how long it took to execute the command.
+
+To make a timer, do the following:
+
+- Make a class with timer functionality that implements the ``CommandListener`` interface.
+- Add an instance of the new class that implements ``CommandListener`` to a ``MongoClientSettings`` object.
+- Configure a ``MongoClient`` instance with the ``MongoClientSettings`` object.
+
+The following code defines the ``CommandTimer`` class which implements the
+``CommandListener`` interface:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/Monitoring.java
+   :language: java
+   :dedent:
+   :start-after: start command-listener-impl
+   :end-before: end command-listener-impl
+
+The following code adds an instance of the ``CommandTimer`` class to a
+``MongoClientSettings`` object, and configures a ``MongoClient`` instance with the
+``MongoClientSettings`` object. The code then runs some database commands to test the
+timer.
+
+.. literalinclude:: /includes/fundamentals/code-snippets/Monitoring.java
+   :language: java
+   :dedent:
+   :start-after: start monitor-command-example
+   :end-before: end monitor-command-example
+
+The above code snippet should produce output that looks like this:
+
+.. code-block:: none
+   :copyable: false
+
+   Command 'find' took 19 milliseconds
+   Command 'aggregate' took 15 milliseconds
+   Command 'endSessions' took 14 milliseconds
+
+For more information on the classes and methods mentioned in this section, see
+the following API documentation:
+
+- :java-docs:`CommandListener <apidocs/mongodb-driver-core/com/mongodb/event/CommandListener.html>`
+- :java-docs:`MongoClientSettings <apidocs/mongodb-driver-core/com/mongodb/MongoClientSettings.html>`
+- :java-docs:`MongoClient <apidocs/mongodb-driver-sync/com/mongodb/client/MongoClient.html>`
+- :java-docs:`CommandStartedEvent <apidocs/mongodb-driver-core/com/mongodb/event/CommandStartedEvent.html>`
+- :java-docs:`CommandSucceededEvent <apidocs/mongodb-driver-core/com/mongodb/event/CommandSucceededEvent.html>`
+- :java-docs:`CommandFailedEvent <apidocs/mongodb-driver-core/com/mongodb/event/CommandFailedEvent.html>`
+
+Cluster Events
+~~~~~~~~~~~~~~
+
+A cluster event in the driver corresponds to a change in the state of the
+MongoDB instance or cluster you have connected the driver to. Cluster events are
+also called server discovery and monitoring (SDAM) events. 
+ 
+The driver defines nine cluster events. The driver divides these nine events
+between three separate listener interfaces which each listen for three of the
+nine events:
```

Comment:
```suggestion
The driver defines nine cluster events and divides them into three listener interfaces. Each interface listens for three of the nine events:
```

Reply:
I am not sure how this edit improves the piece. This seems to duplicate "interface" rather than "the driver" in breaking this across multiple sentences. 

Am I missing the crux of this edit in some way?